### ISSUE TITLE
[Steam Deck] No video output on Gaming Mode

### ISSUE THREAD

**Issue by MayorRev** [maltejur/discord-screenaudio#75]
Been trying this out on the Steam Deck and it works great on desktop mode, but unfortunately a lot of the games I want to screenshare don't work on desktop mode, but _do_ work with Gaming mode.  Unfortunately when attempting to SS in it, only audio is output and the stream is black.

**Comment by maltejur** [comment]
Hm sadly I do not have a Steam Deck and can't test this. But can the normal discord client/browser version stream on the Steam Deck?

**Comment by MayorRev** [comment]
> Hm sadly I do not have a Steam Deck and can't test this. But can the normal discord client/browser version stream on the Steam Deck?

The Discord app available on the deck's Discover app, works fine for the most part on both gaming and desktop mode, but it lacks audio when screensharing.

**Comment by Maciejka1** [comment]
Same on my deck

**Comment by brendenhoffman** [comment]
@maltejur I hope I can shed some light on this. The reason the screen is black is simply because this client doesnt allow you to choose the video source. The way the Steam Deck uses gamescope, it renders everything per window, but the capture is just a black "desktop".

Is there a reason this client can't choose the video source like the normal client?

**Comment by maltejur** [comment]
> Is there a reason this client can't choose the video source like the normal client?

It is a general limitation of QtWebEngine, which is what discord-screenaudio is powered by (see #1). That you are able to select the window on Wayland is only thanks to a "hack" enabling the usage of desktop portals.

As far as I know, the game mode on the steam deck is using wayland though. So this probably really is a problem with the xdg-desktop-portal. Does [doing this](https://steamcommunity.com/app/1675200/discussions/2/3267932441830585132/#c3269060419611556415) change anything?

And the desktop portal is probably now showing up in desktop mode because that is using X11, and thus cant use portals.

**Comment by brendenhoffman** [comment]
No, I can't install xdg-desktop-portal-kde due to it pulling the whole plasma package group (or without disabling read only), so setting XDG_CURRENT_DESKTOP=KDE doesn't work. However, if I could leverage a different portal such as xdg-desktop-portal-gtk or xdg-desktop-portal-wlr that would work, I have those installed. I don't know the requirements to launch any of the desktop portals applications though, especially for gamescope.

I don't see why this wouldn't be possible though, even if it requires you to have a chroot environment set up to install packages.

**Comment by TauAkiou** [comment]
it's probably more likely that Gamescope's current lack of support for Wayland clients is the culprit here, since any applications currently run in X11 mode where portals don't apply: https://github.com/Plagman/gamescope/issues/543

**Comment by brendenhoffman** [comment]
I see. I mean, there is still the issue of installing a usuable portal, but you might be right.

**Comment by TauAkiou** [comment]
my guess is that the Gamescope issue will be resolved at around the same time Valve decides migrates desktop mode over to Wayland, in which case they'll supply xdg-desktop-portal-kde by default.

**Comment by shigutso** [comment]
Any workarounds found while a fix is not possible?

**Comment by elheremes** [comment]
Any workarounds? The default discord screen sharing works in gaming mode, but the discord-screenaudio sharing don't. Works flawless on desktop mode.

**Comment by SSUPII** [comment]
Desktop mode with the Flatpak works extremely well, but in my case in Game Mode the app doesn't launch at all spinning indefinitely.

**Comment by IsaacFehr** [comment]
Has anyone found a workaround for getting video to come through in gaming mode?

I'm seeing the same problem of not being able to choose a window as described above, and sharing the desktop just showing a black screen.

As a note: Vesktop and Webcord (installed through flatpak) can also share audio, but they can select a game window, even if it still shows up all black.  

The official Discord client can share a game window in gaming mode, just without audio.  

What is the difference between the clients that is causing this?

**Comment by 5310** [comment]
I just switched to Desktop mode from a Gamemode Tabletop Simulator (native)
session where I was using Vesktop to share the game. Have used it with
Balatro (Proton) as well, in recent weeks. Both video and audio worked fine
for me here!
